### ISSUE TITLE
An ExperimentTracker port, so a run records what only the service knows

### ISSUE THREAD

**Issue by pergazuz** [pergazuz/LB-anythings#13]
## What to build

An `ExperimentTracker` port, so that a Training Run records the things only the service knows,
and the Operator can plot detector quality against how much labelling has been done.

The MLflow slice records what ultralytics knows: hyperparameters, per-epoch losses, mAP.
Ultralytics cannot know the rest, because the rest lives in the application ring:

- how many Examples the Training Set held when the run started, and how they split;
- whether the run came from the Retrain Threshold, the Start Training button, or the CLI;
- for a Start Training run, how many Tasks the project export yielded and how many were
  unannotated or unreadable;
- the Checkpoint version the run produced, and the version it superseded;
- the class names in play.

The first of those is the important one. *Is labelling still paying off?* is answered by mAP
plotted against Training Set size, and only the application ring holds the size. Without it
the MLflow runs are a list of numbers with no x-axis worth having.

## Acceptance criteria

- [ ] `ExperimentTracker` is declared as a Protocol in `application/ports.py` alongside the
      others, in domain vocabulary, and knows nothing of MLflow.
- [ ] The default implementation is a no-op, so the service runs and every test passes with no
      tracker configured and no mlflow installed.
- [ ] An MLflow adapter lives at `adapters/outbound/mlflow/`, and the import-linter contract
      that keeps adapters independent of one another is extended to name it.
- [ ] The use cases that launch a run record the facts above through the port; the trainer
      adapter does not, because it does not know them.
- [ ] A run started by the Retrain Threshold and one started by Start Training are
      distinguishable in the recorded run, and each carries the Training Set size at launch.
- [ ] The recorded run ties to the ultralytics one from the MLflow slice, so the metrics and
      these facts are one row and not two.
- [ ] Contract test for the adapter, and use-case tests driving a fake tracker through the HTTP
      seam like every other port.
- [ ] Verified for real: two runs at different Training Set sizes, and the mAP-against-size
      pair readable from the recorded runs.

## Blocked by

The MLflow slice.

## Out of scope

- Reporting or charting inside this service. MLflow's own UI is the surface.
- Recording anything per Annotation or per Prediction. This is about Training Runs.

**Comment by pergazuz** [comment]
Done in 14dd5c2.

`ExperimentTracker` is a Protocol beside the other ports with a no-op default, so the service and every test run untracked. `LaunchFacts` carries the trigger, the Training Set size, the Checkpoint the run is trying to beat, and for Start Training what the export held.

## One row, not two

The launcher opens the recorded run, logs its facts and **closes it**, then hands the id down as `--tracked-as`; the Training Run puts it in `MLFLOW_RUN_ID` and ultralytics' callback resumes it. Closing rather than leaving it open matters: a Training Run whose process dies would otherwise leave a run marked running for ever, and MLflow resumes a finished run happily.

## Verification

Three real Training Runs through the live server:

```
active-...T041217Z  trigger=retrain-threshold  size=21  mAP50-95=0.2607  serving=best.pt@20260630T091311Z
active-...T041255Z  trigger=start-training     size=27  mAP50-95=0.5561  exported=7

mAP against Training Set size:
   21 Examples -> 0.2607
   27 Examples -> 0.5561
```

That table is the deliverable. The annotation between those two was refused — `task 7002: ... no Training Run (a Training Run is already active)` — which is the concurrency guard working, not a lost record; my poll raced the run starting. No stray `./mlruns`.

218 passed / 1 skipped, mypy clean, ruff clean, 4 import contracts kept.

## Deviations from the ticket

- **All MLflow knowledge moved into `adapters/outbound/mlflow/`**, further than the ticket asked. The tracker needed `TrackingConfig` and `ensure_experiment`, which #12 had put in the yolo adapter — importing them would have broken the adapter-independence contract. The yolo adapter keeps only the ability to drop ultralytics' mlflow callbacks from a model instance, and the composition root sets the environment the callback reads.
- **Class names are not recorded.** `ExampleStore` does not expose them through the port, only the filesystem adapter does, and widening a port for one parameter was not worth it.
- **The Checkpoint the run produced is not recorded as a launch fact** — it cannot be, it does not exist yet. Ultralytics already archives it as an artifact on the same row, so the pair is there.
- **`CONTEXT.md` gained Run Trigger, and its Retrain Threshold entry was wrong**: it said the Training Set must *grow by* the threshold, when the rule is a multiple of its size. Same error the README had. Corrected.
